### PR TITLE
YIQ-based color metric

### PR DESCRIPTION
**Pull request by mourner**
Implements the formula [in this paper](http://www.progmat.uaem.mx:8080/artVol2Num2/Articulo3Vol2Num2.pdf), which is much closer to CIEDE2000 in quality unlike YUV distance which was non-uniform.

Also fixes RGBA to RGB conversion to blend with white rather than black.

- [x] update tests
- [ ] try to optimize some more
- [x] cleanup and code comments
- [x] update readme

cc @hastebrot

### CODE REVIEW

**Comment by mourner** [comment]
Actually there's one more fast distance metric to check out: http://www.compuphase.com/cmetric.htm
I'm eager to do a comparison of Kotsarenko-Ramos YIQ vs Riemersma RGB vs CIEDE2000 vs RGB both in terms of perceptual difference and performance.

**Comment by hastebrot** [comment]
The paper's algorithm uses sRGB color space for calculations. It explicitly uses R'G'B' instead of RGB. The prime `'` signifies that the quantities are gamma-corrected [1]. Both [1] and [2] (under `C_srgb´) give simple formulas to convert from RGB to sRGB.

I don't know if this matters for color differences. I also don't know if the test PNG files are in RGB or sRGB; my image viewer say RGB. I also read somewhere, that JPEG images of digital cameras are normally in sRGB.

[1] http://www.getreuer.info/home/colorspace#TOC-Gamma-Correction
[2] https://en.wikipedia.org/wiki/SRGB#The_forward_transformation_.28CIE_xyY_or_CIE_XYZ_to_sRGB.29

**Comment by hastebrot** [comment]
Nevermind. The images are in sRGB. I've checked the fixture PNG images with ImageMagick's `identify`:

~~~
$ identify 1a.png
1a.png PNG 512x256 512x256+0+0 8-bit sRGB 30.9KB 0.000u 0:00.001

$ identify 2a.png
2a.png PNG 256x256 256x256+0+0 8-bit sRGB 256c 33KB 0.000u 0:00.000
~~~

**Comment by mourner** [comment]
Just looked it up and Web technologies like Canvas are also using sRGB: https://html.spec.whatwg.org/multipage/infrastructure.html#colours

**Comment by mourner** [comment]
Played with some different metrics here: http://bl.ocks.org/mourner/8a3968f74da743a754af

File: index.js [hunk 1]
```diff
@@ -8,8 +8,9 @@ function pixelmatch(img1, img2, output, width, height, options) {
 
     var threshold = options.threshold === undefined ? 0.1 : options.threshold;
 
-    // maximum acceptable square YUV distance between two colors
-    var maxDelta = 255 * 255 * 3 * threshold * threshold,
+    // maximum acceptable square distance between two colors;
+    // 35215 is the maximum possible value for the YIQ difference metric
+    var maxDelta = 35215 * threshold * threshold,
```
Comment (hastebrot) [inline index.js]: What was the formula to get `35215`?

Comment (mourner) [inline index.js]: No formula, just a loop through RGB values like you did for YUV and a round :) AFAIR it's also the same value you get for squared distance between [255, 0, 0] and [0, 255, 255].

Comment (hastebrot) [inline index.js]: I get following results (https://github.com/hastebrot/notebooks/blob/master/ipython/yiq-color-range.ipynb):

~~~
minmax(y): 0.0 255.00000255
minmax(i): -151.97438745 151.97438745
minmax(q): -133.26736305 133.26736305
~~~

~~~
dist(min, max): 272.7322528519619
dist(red, cyan): 187.6559323748062
~~~

>AFAIR it's also the same value you get for squared distance between [255, 0, 0] and [0, 255, 255].

Yes, red and cyan have the largest color distance. But, I used the weighted euclidian distance from `colorDelta()` with red and cyan and got different results. :confused:

Comment (hastebrot) [inline index.js]: Ahh, got it. Perfect. It is 35214.74895545784 ≈ 35215 without taking the root. Sorry for the noise!

Comment (mourner) [inline index.js]: Maybe an error in your calculations somewhere. I get `35214.74895545785` both as max and squared distance between red and cyan.

**Pull request merged**